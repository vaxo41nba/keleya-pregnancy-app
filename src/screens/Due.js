import React, { useState } from 'react';
import {
  Image,
  Pressable,
  ScrollView,
  StyleSheet,
  Text,
  View,
} from 'react-native';
import DateTimePicker from '@react-native-community/datetimepicker';
import { t } from 'i18next';

import BackArrow from '../components/BackArrow';
import AppButton from '../components/AppButton';
import images from '../constants/images';
import colors from '../constants/colors';

export default function Name({ navigation, route }) {
  const [date, setDate] = useState(new Date());
  const [datePickerVisible, setDatePickerVisible] = useState(false);

  const onChange = (event, selectedDate) => {
    setDatePickerVisible(false);
    setDate(selectedDate);
  };

  const formatDate = () => {
    const selectedDate = new Date(date);
    const array = selectedDate.toString().split(' ');
    return `${array[1]} ${array[2]}, ${array[3]}`;
  };

  const navigate = () => navigation.navigate('Workout');

  return (
    <ScrollView
      style={styles.container}
      contentContainerStyle={{
        minHeight: '100%',
        paddingBottom: 30,
      }}
    >
      <View style={styles.header}>
        <Image source={images.due} style={styles.image} />
        <BackArrow navigation={navigation} />
      </View>

      <Text style={styles.h1}>
        {t('When is your baby due,')} {route?.params?.name?.trim()}?
      </Text>

      <Pressable
        style={styles.datePicker}
        onPress={() => setDatePickerVisible(true)}
      >
        <Text style={{ color: '#5294fc' }}>{formatDate()}</Text>
      </Pressable>
      {datePickerVisible && (
        <DateTimePicker
          testID="dateTimePicker"
          value={date}
          mode="date"
          onChange={onChange}
        />
      )}

      <View style={styles.footer}>
        <AppButton text={t('Continue')} onPress={navigate} />
      </View>
    </ScrollView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: colors.WHITE,
  },
  datePicker: {
    width: 100,
    height: 35,
    borderRadius: 5,
    backgroundColor: '#eeeeef',
    alignSelf: 'center',
    alignItems: 'center',
    justifyContent: 'center',
  },
  footer: {
    flex: 1,
    justifyContent: 'flex-end',
    alignItems: 'center',
    marginTop: 30,
  },
  header: { width: '100%', maxHeight: 550, marginBottom: -80 },
  h1: {
    maxWidth: '85%',
    fontSize: 25,
    textAlign: 'center',
    alignSelf: 'center',
    color: colors.GREYISH_BROWN,
    marginBottom: 20,
  },
  image: { width: '100%', height: '100%' },
});
