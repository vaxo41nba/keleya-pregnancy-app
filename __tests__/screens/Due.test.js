import React from 'react';
import renderer from 'react-test-renderer';
import Due from '../../src/screens/Due';

test('renders correctly', () => {
  const tree = renderer.create(<Due />).toJSON();
  expect(tree).toMatchSnapshot();
});
