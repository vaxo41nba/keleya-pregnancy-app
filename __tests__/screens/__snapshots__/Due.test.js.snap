// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  contentContainerStyle={
    {
      "minHeight": "100%",
      "paddingBottom": 30,
    }
  }
  style={
    {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "marginBottom": -80,
          "maxHeight": 550,
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          {
            "height": "100%",
            "width": "100%",
          }
        }
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "height": 35,
            "justifyContent": "center",
            "left": 35,
            "opacity": 1,
            "position": "absolute",
            "top": 54,
            "width": 40,
          }
        }
      >
        <Image
          source={1}
          style={
            {
              "height": 15,
              "width": 20,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        {
          "alignSelf": "center",
          "color": "#4a4a4a",
          "fontSize": 25,
          "marginBottom": 20,
          "maxWidth": "85%",
          "textAlign": "center",
        }
      }
    >
       
      ?
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#eeeeef",
          "borderRadius": 5,
          "height": 35,
          "justifyContent": "center",
          "width": 100,
        }
      }
    >
      <Text
        style={
          {
            "color": "#5294fc",
          }
        }
      >
        Aug 29, 2023
      </Text>
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "flex-end",
          "marginTop": 30,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#69c0ba",
            "borderRadius": 10,
            "height": 55,
            "justifyContent": "center",
            "opacity": 1,
            "paddingHorizontal": 10,
            "width": "80%",
          }
        }
      >
        <Text
          style={
            {
              "color": "#ffffff",
              "fontSize": 22,
              "textAlign": "center",
            }
          }
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;
